### DISCUSSION TITLE
wsdl  no run   revokeCert (soap xml)

### DISCUSSION THREAD

**Discussion by jpertuz-19** [Keyfactor/ejbca-ce#570] [General]
Hi.   

i use SOAP  xml  for create user/certificate.        all work fine. 

                  <ws:editUser>\n' +
        '         <arg0>\n' +
        '            <caName>ManagementCA</caName>\n' +
       '            <certificateProfileName>ENDUSER</certificateProfileName>\n' +    
                 bla bla bla 
       '            <username>test</username>\n' +
        '         </arg0>\n' +
        '      </ws:editUser>\n' +

I can revoker the entity with revokeUser but with revokeCert for the certificate I get an error.

java.lang.IllegalArgumentException: badly formatted directory string

<ws:revokeCert>\n' +
        '            <arg0>CN=MYAC</arg0>\n' +                      "Issuer"
        '            <arg1>123456789123456</arg1>\n' +        "certificateSN"
        '            <arg2>20</arg2>\n' +                                     "reason"   ????  dude 
        '      </ws:revokeCert>\n' +

The arg3 "reason" I can't figure out what code to put in, it's the one that gives the error.

**Comment by jpertuz-19** [comment]
Solved:

<ws:revokeCert\n' +
' CN=MYAC,C=VE\n' + "Issuer"
' 123456789123456\n' + "certificateSN"
' 1\n' + "reason" ???? dude
' </ws:revokeCert>\n' +

Can you help me print a user's certificateSN?         findUser? or findCerts?

**Comment by jpertuz-19** [comment]
For revoke,   i use   firts    findCerts,    for  obtain cert.     then  read x509 extrack   serial-number

with serialnumber  use  rovokeCert

**Reply by surtan02** [reply]
hello sorry if it not related, but can u give me brief example of ur editUser request? I am using a EJBCA-CE with Docker but cant send a request to editUser WS.

My request like this
<?xml version="1.0" encoding="utf-8"?>
<soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">
  <soap:Body>
    <editUser xmlns="http://ws.protocol.core.ejbca.org/">
      <arg0>
        <username>soapUser</username>
        <password>superadmin</password>
        <clearPwd>true</clearPwd>
        <subjectDN>cn=soap-user</subjectDN>
        <caName>CloudCA</caName>
        <email>soap@gmail.com</email>
        <status>10</status>
        <tokenType>P12</tokenType>
        <sendNotification>false</sendNotification>
        <keyRecoverable>false</keyRecoverable>
        <endEntityProfileName>MYPROFILE</endEntityProfileName>
        <certificateProfileName>ENDUSER</certificateProfileName>
        <hardTokenIssuerName>null</hardTokenIssuerName>
      </arg0>
    </editUser>
  </soap:Body>
</soap:Envelope>

**Reply by primetomas** [reply]
Sounds like a new issue?
I recommend that you do it with clientToolBox, and debug the SOAP message. Then you will see exactly how it will look like.
https://doc.primekey.com/ejbca/ejbca-operations/ejbca-ca-concept-guide/protocols/web-service-interface#WebServiceInterface-WebServiceMessageDebuggingWS_message_debugging

**Reply by surtan02** [reply]
solved it, I made syntax mistake in the xml body. thanks for the response

**Reply by jpertuz-19** [reply]
<ws:editUser>
         <arg0>
            <caName>AdministracionCA</caName>
            <certificateProfileName>ENDUSER</certificateProfileName>
            <clearPwd>true</clearPwd>
            <email>usuario@domainexam.com</email>
            <endEntityProfileName>EMPTY</endEntityProfileName>
            <keyRecoverable>true</keyRecoverable>
            <password>xxxalgo123</password>
            <sendNotification>false</sendNotification>
            <status>10</status>
            <subjectDN>cn=Jose</subjectDN>
            <tokenType>P12</tokenType>
            <username>Jose</username>
         </arg0>
      </ws:editUser>
